ChangeLog.d/add-aes-hardware-only-option.txt

```diff
@@ -0,0 +1,6 @@
+Features
+   * New configuration option MBEDTLS_AES_USE_HARDWARE_ONLY introduced. When
+     using CPU-accelerated AES (e.g., Arm Crypto Extensions), this option
+     disables the plain C implementation and the run-time detection for the
+     CPU feature, which reduces code size and avoids the vulnerability of the
+     plain C implementation.
```

Comment:
If the ChangeLog is information for library users, this doesn't really tell me whether I need it or not. Should we add something like

`Enable if - but only if - you know the library will be used on systems with CPU-accelerated AES`

And I wonder if a sentence like that might be useful in the config file too?

Reply:
Agree with Dave. The information should be put at document of option.

Reply:
It's just that, reading both the ChangeLog entry and the config file entry after being away from it for a few days, neither gives me a sense of when I might want to enable it. The ChangeLog entry says what happens, but doesn't give reasons why enabling it might be a good idea.

Reply:
Surely "reduces code size and avoids the vulnerability" tells users why they might want it?